devito/arch/archinfo.py

```diff
@@ -216,13 +216,36 @@ def lspci_gpu_info(text):
                 gpu_infos.append(gpu_info)
         return filter_real_gpus(gpu_infos)
 
+    def nvidiasmi_gpu_info(text):
```

Comment:
Why not use the nvidia-smi python package? 
https://pypi.org/project/nvidia-ml-py3/

May be overkill I guess but may be safer.

Reply:
but a compulsory dependence? 

I'd say no, and then what if one runs w/o it? do we ask him to install it raising a warning?

Reply:
Can `nvaccelinfo` do the work as well ?

Reply:
Yes, but it requires loading SDK environment

Reply:
let's stick to the current approach. My workstation has nvidia-smi from 5 years ago and the output is in the same format as today's version. We'll revisit and consider adding that optional dependency in the future if ever necessary